scrnaseq.Rmd

```diff
@@ -2189,8 +2183,50 @@ output_files[["degs_enrichr"]] = purrr::map_chr(degs, function(degs_subsets) {
 Misc(sc, "degs") = degs
 ```
 
+```{r degs_plots}
+for (i in seq(degs)) { 
+  for (j in seq(degs[[i]])) {
+
+    # Average expression of all genes
+    x = subset(sc, cells=degs[[i]][[j]]$cells_group2_idx) %>% GetAssayData(slot="data")
```

Comment:
You say that, because Seurat excludes genes that don't make `min.pct` or `logfc.threshold`? Yes, I'd still plot all genes -> it gives an overall impression of how different the two groups are, and in fact, all genes have been looked at, no? If they are grey, they either don't make `min_pct` or `logfc.threshold`, or are not significant.

Reply:
If you find this unclear, we can add a sentence to the plots.

Reply:
No Seurat only tests genes which meet the min.pct, logfc.threshold, ... criteria since it is faster (also check the Seurat DE vignette). If you want to show all the genes, they should all be tested (which will slow down the script). Alternatively one could think about another color for "not tested" but that is also suboptimal.

Reply:
I think we should only show the genes that got tested.

Reply:
As we discussed offline, all genes are "touched" once in the function. That is, if they pass `min.pct` and `logfc.threshold`, they are tested and get a p-value. So if a gene is visualised in grey, it means that it either does not make the first two cutoffs, or the p-value is too high. In either case, only coloured genes make all cutoffs.

Reply:
ok